portal/static/js/flask_user/register.js

```diff
@@ -0,0 +1,119 @@
+                self.initVis();
+                self.initPasswordPopover();
+                self.initFieldEvents();
+            });
+        };
+        this.initVis = function() {
+            // Need to properly form attribute name
+            $("[data-match_error]").each(function(){
+                $(this).attr("data-match-error", $(this).attr("data-match_error"));
+            });
+            $("input[type='submit']").addClass("disabled");
+        };
+        this.checkValidity = function(submit) {
+            var allowed = true;
+            ["#email", "#password", "#retype_password"].forEach(function(fieldId) {
+                if (!$(fieldId).val()) {
+                    allowed = false;
+                }
+            });
+            if (allowed) {
+                allowed = !($(".form-group.has-error").length);
+            }
+            if (allowed) {
+                $("input[type='submit']").removeClass("disabled");
+            } else {
+                $("input[type='submit']").addClass("disabled");
+            }
+            if (!submit) {
+                return;
+            }
+            $("input[type='submit']").trigger("click");
+        };
+        this.initFieldEvents = function() {
+            var self = this;
+            $("input.input-field").each(function() {
+                $(this).on("blur, change", function() {
+                    setTimeout(function() { self.checkValidity(); }, 350);
+                });
+                $(this).on("keyup", function(e) {
+                    e.preventDefault();
+                    setTimeout(function() { self.checkValidity(e.keyCode === 13); } , 350);
+                });
+            });
+            $("#retype_password").on("keyup change", function() {
+                if (!$(this).val()) {
+                    $("#errorretype_password").addClass("has-error").text(i18next.t("Please re-type your password."));
+                } else {
+                    $("#errorretype_password").text("").removeClass("has-error");
+                    self.checkValidity();
+                }
+            });
+            $("#email").on("change", function() {
+                $("#erroremail").text("");
+            });
+
+        };
+        this.setPopoverVis = function(condition, field) {
+            if (condition) {
+                field.removeClass("fail-text").addClass("success-text");
+            } else {
+                field.removeClass("success-text").addClass("fail-text");
+            }
+        };
+        this.initPasswordPopover = function() {
+            var self = this;
+            $("#password").popover({"content": $("#popoverContent").html(), "placement": "top", "html": true, "template": "<div class=\"password-popover popover\" role=\"tooltip\"><div class=\"arrow\"></div><h3 class=\"popover-title\"></h3><div class=\"popover-content\"></div></div>', \"container\": \".password-wrapper\", \"trigger\": \"manual\""}).click(function(e) {
```

Comment:
Is there a way to keep this HTML in the template and just toggle the visibility?

Reply:
done, thank you